File: frontend/src/services/product.service.ts
import api from '@/lib/api';
import { Product } from '@/types';

export const ProductService = {
    getAllProducts: async (category?: string) => {
        const params = category ? { category } : {};
        const response = await api.get<Product[]>('/products', { params });
        return response.data;
    },

    getProductById: async (id: string) => {
        const response = await api.get<Product>(`/products/${id}`);
        return response.data;
    },

    searchProducts: async (query: string) => {
        const response = await api.get<Product[]>('/products/search', { params: { query } });
        return response.data;
    },

    createProduct: async (product: Partial<Product>) => {
        const response = await api.post<Product>('/products', product);
        return response.data;
    },

    updateProduct: async (id: string, product: Partial<Product>) => {
        const response = await api.put<Product>(`/products/${id}`, product);
        return response.data;
    },

    deleteProduct: async (id: string) => {
        const response = await api.delete<{ message: string }>(`/products/${id}`);
        return response.data;
    }
};

File: backend/src/main/java/com/sutrini/service/ProductService.java
package com.sutrini.service;

import com.sutrini.model.Product;
import java.util.List;
import java.util.Optional;

public interface ProductService {
    List<Product> getAllProducts(String category);

    List<Product> searchProducts(String query);

    Optional<Product> getProductById(String id);

    Product createProduct(Product product);

    Product updateProduct(String id, Product product);

    void deleteProduct(String id);
}
